### DISCUSSION TITLE
[Front End] Next.js 13+ & Server Components & Layouts & Streaming

### DISCUSSION THREAD

**Discussion by Dima-Abramenko** [realworld-apps/realworld#1082] [WIP Implementations]
Hello there! 👋

I've started working on Front End implementation of RealWorld app: [repository](https://github.com/Dima-Abramenko/next-layouts-realworld-example-app)

**Motivation:**

Vercel team introduced brand-new approach of building Next.js applications in theirs 13-th major version ([announcement](https://nextjs.org/blog/next-13)). This upgrade brings React 18 features like Server Components, Suspence, Server HTML Streaming and so on. This approach is quite different from what we used before in Next.js.

Therefore I'm planning to implement RealWorld from scratch using this new paradigm 🙂

**Comment by geromegrignon** [comment]
Hi Dmitry, great idea. I heard about them, and they'll be a great addition to existing implementations!

**Comment by kamalbuilds** [comment]
Hi @Dima-Abramenko , this sounds great . I have been building with nextjs from an year now and would like to help you in this WIP .
Can We collaborate ?

**Reply by Dima-Abramenko** [reply]
Hey @legendarykamal ! :)

I'm really appreciate this, I'm building RealWorld app with Next.js 13+ to play with its new concepts and approaches. But first I want to figure out with theirs pros & cons myself :) 

Could I reach out to you a bit later to review my implementation and, if necessary, discuss further steps to improve it? :)

**Reply by kamalbuilds** [reply]
Yes @Dima-Abramenko that sounds nice. Anytime let's connect 🙂

**Comment by leerob** [comment]
This sounds amazing! Happy to help review code.

**Reply by Dima-Abramenko** [reply]
Thank you so much @leerob, I appreciate this! :)

As Next.js "app" directory is still in beta - I'll definitely need a review of my updates for some new concepts :)

**Comment by jilv220** [comment]
My test implementation of realworld in nextjs 13.  
https://next13-realworld.vercel.app/

**Reply by siam-craft** [reply]
@jilv220 hello.. How can I find your code repository? can your share your code here?
